helix-view/src/editor.rs

```diff
@@ -398,18 +398,105 @@ impl Editor {
             .selections
             .remove(&id);
 
-        if close_buffer {
-            // get around borrowck issues
-            let doc = &self.documents[&view.doc];
+        self.tree.remove(id);
+        self._refresh();
+    }
+
+    pub fn close_document(&mut self, id: DocumentId, force: bool) -> anyhow::Result<()> {
+        if !self.documents.contains_key(&id) {
+            anyhow::bail!("document does not exist");
+        }
+
+        log::error!("remove: {:?}", &id);
+        let doc = &self.documents[&id];
+
+        if !force && doc.is_modified() {
+            anyhow::bail!(
+                "buffer {:?} is modified",
+                doc.relative_path()
+                    .map(|path| path.to_string_lossy().to_string())
+                    .unwrap_or_else(|| "[scratch]".into())
+            );
+        }
+
+        if let Some(language_server) = doc.language_server() {
+            tokio::spawn(language_server.text_document_did_close(doc.identifier()));
+        }
+
+        let old_focus = self.tree.focus;
+        let view_ids = self
+            .tree
+            .traverse()
+            .filter(|(_, view)| view.doc == id)
+            .map(|(id, _)| id)
+            .collect::<Vec<_>>();
```

Comment:
You should just iterate over all the views in the tree, a view might have the current document ID in it's `last_accessed_doc` or `jumps` even if it's currently showing a different view.

```rust
for view in tree.views_mut() {
  // call view.jumps.remove(doc_id)
  // clear out the last_accessed_doc if it contains doc_id
  
  // if view.doc == doc_id, do additional logic
}